examples/pipelines/fl_cross_silo_native/submit.py

```diff
@@ -204,93 +205,89 @@ def fl_cross_silo_internal_basic():
-                iteration_name=f"Iteration-{iteration}",
-            )
-            # add a readable name to the step
-            silo_training_step.name = f"silo_{silo_index}_training"
-
-            # make sure the compute corresponds to the silo
-            silo_training_step.compute = silo_config.compute
-
-            # make sure the data is written in the right datastore
-            silo_training_step.outputs.model = Output(
-                type=AssetTypes.URI_FOLDER,
-                mode="mount",
-                path=custom_fl_data_path(
-                    # IMPORTANT: writing the output of training into the orchestrator datastore
-                    YAML_CONFIG.federated_learning.orchestrator.datastore,
-                    f"model/silo{silo_index}",
-                    iteration_num=iteration,
-                ),
-            )
-
-            # each output is indexed to be fed into aggregate_component as a distinct input
-            silo_weights_outputs[
-                f"input_silo_{silo_index+1}"
-            ] = silo_training_step.outputs.model
-
-        # aggregate all silo models into one
-        aggregate_weights_step = aggregate_component(**silo_weights_outputs)
-        # this is done in the orchestrator compute
-        aggregate_weights_step.compute = (
-            YAML_CONFIG.federated_learning.orchestrator.compute
+        ################
+        ### TRAINING ###
+        ################
+
+        # we're using training component here
+        silo_training_step = training_component(
+            # with the train_data from the pre_processing step
+            train_data=silo_pre_processing_step.outputs.processed_train_data,
+            # with the test_data from the pre_processing step
+            test_data=silo_pre_processing_step.outputs.processed_test_data,
+            # and the checkpoint from previous iteration (or None if iteration == 1)
+            checkpoint=running_checkpoint,
+            # Learning rate for local training
+            lr=YAML_CONFIG.training_parameters.lr,
+            # Number of epochs
+            epochs=YAML_CONFIG.training_parameters.epochs,
+            # Dataloader batch size
+            batch_size=YAML_CONFIG.training_parameters.batch_size,
+            # Silo name/identifier
+            metrics_prefix=silo_config.compute,
+            # Iteration number
+            iteration_num=iteration,
         )
         # add a readable name to the step
-        aggregate_weights_step.name = f"iteration_{iteration}_aggregation"
+        silo_training_step.name = f"silo_{silo_index}_training"
+
+        # make sure the compute corresponds to the silo
+        silo_training_step.compute = silo_config.compute
 
         # make sure the data is written in the right datastore
-        aggregate_weights_step.outputs.aggregated_output = Output(
-            type=AssetTypes.URI_FOLDER,
+        silo_training_step.outputs.model = Output(
+            type=AssetTypes.CUSTOM_MODEL,
```

Comment:
Are we sure we want to use CUSTOM_MODEL here? I thought that was triggering the registration of the model somewhere else, which we don't want. (sorry if I'm late to the party and this has been dealt with already)

Reply:
my bad! That should be a `uri_folder`